### ISSUE TITLE
Table.where() with start but no stop & step has surprising behavior

### ISSUE THREAD

**Issue by kanak** [PyTables/PyTables#797]
https://github.com/PyTables/PyTables/blob/master/tables/leaf.py#L442

In that case, it will only query in [start, start +1) i.e. the start row. Based on python indexing behavior, I would expect it to search in [start, nrows).

Here's a code snippet that illustrates the problem:
```python
# Table has an "epoch" field. Row 13969 has the epoch that is used in condition.
# I'm trying to find the next row that has a different epoch.

[(x.nrow, x["epoch"]) for x in table.where("(epoch > 1584698775894018117)", stop=13970)]
# [(13968, 1584698775894024479), (13969, 1584698775895064429)]

[(x.nrow, x["epoch"]) for x in table.where("(epoch > 1584698775894018117)", start=13965, stop=13970)]
# [(13968, 1584698775894024479), (13969, 1584698775895064429)]

[(x.nrow, x["epoch"]) for x in table.where("(epoch > 1584698775894018117)", start=13965)]
# []
```

**Comment by avalentino** [comment]
@kanak I think that the link the you posted now points to an incorrect location.
Could you please explain better that is the problem and maybe provide a simple code snippet/data file to reproduce the issue?

**Comment by kanak** [comment]
@avalentino : I have updated the link.

I found the current behavior of `where(query, start)` to be surprising because I expected it to search from `start` row till the `end` of the dataset. However, right now it only searches from `start` to `start + 1`.

**Comment by avalentino** [comment]
Thanks @kanak, I understand now.
By the way the function you are pointing to is called `_process_range_read` so I think we cannot change it without breaking something else, but probably we can manage to fix the `where` behavior anyway.

Would you like to submit a PR for this?

**Comment by kanak** [comment]
@avalentino : Since `where` uses this function, I'm concerned that this change would result in a change to public APIs:

https://github.com/PyTables/PyTables/blob/98a56b084b267156cb25e9c0ea8afb466fd70e75/tables/table.py#L1471

That said, I'm happy to work on the PR.

**Comment by avalentino** [comment]
@kanak yes, the idea is to not change `_process_range_read` but rather to modify something around this line https://github.com/PyTables/PyTables/blob/b2d5a930cea0460656e596cde0962b70be87f1e4/tables/table.py#L1471
to call a different function for slice parameter processing (right now I do not remember if there is one) of pre-processing the `stop` parameter to get the correct behavior. Maybe just setting it to -1 could work.

The important part is to check that all tests keep passing or to adjust the test code for the new semantic.
